### PR TITLE
DOC: Added pre-commit link inside the guideline for developers.

### LINKED ISSUE #48941: DOC: Create link from "Creating a development environment" to "pre-commit"
~~I had to recreate my pandas environment from `environment.yml` .   Then when I updated a PR, the pre-commit was failing because `autotyping` was not installed.  It needs to be added to `environment.yml`~~

This is a docs issue.  See https://github.com/pandas-dev/pandas/issues/48941#issuecomment-1270114184

### PR DESCRIPTION
**Pull request by HatimZ**
- [x] closes #48941

Updated the contributor guidelines doc to include the redirection url for pre-commit. 
Help developers to setup the environment.

### CODE REVIEW

**Comment by WillAyd** [comment]
I think rather than creating a new section the idea is to link to the already-existing section on pre-commit in contributing_codebase.rst - can you give that a look? @Dr-Irv @MarcoGorelli

**Comment by Dr-Irv** [comment]
> I think rather than creating a new section the idea is to link to the already-existing section on pre-commit in contributing_codebase.rst - can you give that a look? @Dr-Irv @MarcoGorelli

Yes, I agree.  Just create a link (and make sure you use the short form link, not the http) with text that says "Be sure to also install the pre-commit hooks", where the phrase "pre-commit hooks" links to that section of the docs.

**Comment by HatimZ** [comment]
@Dr-Irv , I have updated the text as per your instructions. Thanks.

File: doc/source/development/contributing_environment.rst [hunk 1]
```diff
@@ -216,6 +216,8 @@ should already exist.
    python setup.py build_ext -j 4
    python -m pip install -e . --no-build-isolation --no-use-pep517
 
+Be sure to also install the `pre-commit hooks <https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#pre-commit>`__.
```
Comment (Dr-Irv) [inline doc/source/development/contributing_environment.rst:219]: I think (not 100% sure) that the link should be of the form ``:ref:`per-commit hooks <contributing.pre-commit>` ``

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: doc/source/development/contributing_environment.rst (hunk 1 above)
Comment (HatimZ) [reply doc/source/development/contributing_environment.rst:219]: When I write :ref:`per-commit hooks <contributing.pre-commit>` , it doesn't show a hyperlink text in the Preview section.
image

Comment (Dr-Irv) [reply doc/source/development/contributing_environment.rst:219]: That's to be expected.  If you look at similar links in that document, e.g., ``:ref:`Continuous Integration <contributing.ci>` ``, you'll see the same thing.

What you should do is build the docs locally and then you can test that the link is correct.

Comment (MarcoGorelli) [reply doc/source/development/contributing_environment.rst:219]: or you can download the artifact from the CI build https://github.com/pandas-dev/pandas/actions/runs/3337146888 (see "artifacts" at the bottom of the page)

**Comment by HatimZ** [comment]
@Dr-Irv . I have made the respective changes.
I understand it now. There are two kinds of links in the file, one with http and the other short form link with ref format. Although both work the same way, the short form link redirects in an offline manner.

 Did I understand it correctly ?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: doc/source/development/contributing_environment.rst [hunk 2]
```diff
@@ -216,6 +216,8 @@ should already exist.
    python setup.py build_ext -j 4
    python -m pip install -e . --no-build-isolation --no-use-pep517
 
+Be sure to also install the :ref:`pre-commit hooks <contributing.pre-commit>`
```
[the remark this answers is not in the document]
Comment (Dr-Irv) [reply doc/source/development/contributing_environment.rst:219]: But if you don't install pre-commit, then it becomes really hard to submit any PR's, right?

[the remark this answers is not in the document]
Comment (HatimZ) [reply doc/source/development/contributing_environment.rst]: Done.

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
Thanks for your PR

Not keen on this location as it's only visible to anyone reading "option 1b"

I'd suggest placing it right after the first paragraph, just after
```
but if you skip creating the development environment you won't be able to build the documentation locally before pushing your changes.
```

I also agree on changing `be sure to`, e.g. to `it's recommended to`

**Comment by HatimZ** [comment]
> Thanks for your PR
> 
> Not keen on this location as it's only visible to anyone reading "option 1b"
> 
> I'd suggest placing it right after the first paragraph, just after
> 
> ```
> but if you skip creating the development environment you won't be able to build the documentation locally before pushing your changes.
> ```
> 
> I also agree on changing `be sure to`, e.g. to `it's recommended to`

Thanks for you help. Applied the changes,

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me

I've put the 1.5.2 milestone as it'd be good to backport this one

**Review by Dr-Irv — APPROVED** [review]

**Pull request merged**